howto/diagnose-with-tracing/open-telemetry-collector.md

```diff
@@ -0,0 +1,254 @@
+      pipelines:
+        traces:
+          receivers: [opencensus]
+          exporters: [azuremonitor/2]
+          processors: [batch, queued_retry]
+---
+apiVersion: v1
+kind: Service
+metadata:
+  name: otel-collector
+  labels:
+    app: opencesus
+    component: otel-collector
+spec:
+  ports:
+  - name: opencensus # Default endpoint for Opencensus receiver.
+    port: 55678
+    protocol: TCP
+    targetPort: 55678
+  selector:
+    component: otel-collector
+---
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: otel-collector
+  labels:
+    app: opentelemetry
+    component: otel-collector
+spec:
+  replicas: 1  # scale out based on your usage
+  selector:
+    matchLabels:
+      app: opentelemetry
+  template:
+    metadata:
+      labels:
+        app: opentelemetry
+        component: otel-collector
+    spec:
+      containers:
+      - name: otel-collector
+        image: otel/opentelemetry-collector-contrib-dev:latest
+        command:
+          - "/otelcontribcol"
+          - "--config=/conf/otel-collector-config.yaml"
+        resources:
+          limits:
+            cpu: 1
+            memory: 2Gi
+          requests:
+            cpu: 200m
+            memory: 400Mi
+        ports:
+          - containerPort: 55678 # Default endpoint for Opencensus receiver.
+        volumeMounts:
+          - name: otel-collector-config-vol
+            mountPath: /conf
+          #- name: otel-collector-secrets
+          #  mountPath: /secrets
+        livenessProbe:
+          httpGet:
+            path: /
+            port: 13133
+        readinessProbe:
+          httpGet:
+            path: /
+            port: 13133
+      volumes:
+        - configMap:
+            name: otel-collector-conf
+            items:
+              - key: otel-collector-config
+                path: otel-collector-config.yaml
+          name: otel-collector-config-vol
+EOF
+
+```
+
+
+Set up a Dapr exporter component
+```
+kubectl apply -f- <<EOF
+apiVersion: dapr.io/v1alpha1
+kind: Configuration
+metadata:
+  name: appconfig
+  namespace: default
+spec:
+  # Configure Dapr tracing.
+  tracing:
+    samplingRate: "1"
+---
+apiVersion: dapr.io/v1alpha1
+kind: Component
+metadata:
+  name: native
+  namespace: default
+spec:
+  type: exporters.native
+  metadata:
+  - name: enabled
+    value: "true"
+  - name: agentEndpoint
+    value: "otel-collector.default.svc.cluster.local:55678"
+EOF
+```
+
+### Deploy your app with tracing
+
+When running in Kubernetes mode, apply the configuration by adding a
+`dapr.io/config` annotation to the container that you want to
+participate in the distributed tracing, as shown in the following
+example:
+
+```yaml
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  ...
+spec:
+  ...
+  template:
+    metadata:
+      ...
+      annotations:
+        dapr.io/enabled: "true"
+        dapr.io/app-id: "calculator-front-end"
+        dapr.io/app-port: "8080"
+        dapr.io/config: "tracing"
```

Comment:
This should be "appconfig" not "tracing" to match the example about, correct? 
This is a how to, so the steps have to all work together.

Reply:
Thanks for fixing this already.